jax/_src/callback.py

```diff
@@ -225,7 +225,7 @@ def _check_shape_dtype(shape_dtype):
   dt = np.dtype(shape_dtype.dtype)
   if dtypes.canonicalize_dtype(dt) != dt:
```

Comment:
Should we check that `jax_enable_x64` is disabled as well or does it follow from `canonicalize_dtype(dt) != dt`?

Reply:
We are using the same pattern elsewhere in the code, to check that x64 is not enabler and the type is a 64-bit type.